### PR TITLE
Instance/newsletter

### PR DESCRIPTION
**Pull request by everyplace**
This PR adds a set of examples for the newsletter api.

> [!Note]
> The code in this PR has been updated to include the final API.

This PR is available for preview [here](https://newsletter-dot-reader-revenue-demo.ue.r.appspot.com/).

In addition, this PR adds a few new capabilities:
- Adds an ability to configure the language parameter in the html template
- Adds a new capability to hide items from the navigation

### CODE REVIEW

File: app/views/layouts/demo-layout.html [hunk 1]
```diff
@@ -111,17 +111,21 @@
       <div class="developer-site-nav">
 
         {{#each data.nav}}
-        <hr />
+          {{#unless this.options.suppressInNav}}
```
Comment (everyplace) [inline app/views/layouts/demo-layout.html:114]: This level of suppression suppresses the entire category from displaying.

File: app/views/layouts/demo-layout.html [hunk 2]
```diff
@@ -111,17 +111,21 @@
...
+          <hr />
           <h3 class="developer-site-nav-header developer-site-nav-title" >{{this.section}}</h3>
           <ul class="developer-site-nav-list">
             {{#each this.links}}
+              {{#unless this.options.suppressInNav}}
```
Comment (everyplace) [inline app/views/layouts/demo-layout.html:119]: This level of suppression only suppresses this item within the category.

File: app/content/newsletter-manual.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+### Handle the response
+
+The `onResult` callback will include information on the configuration used
+to create the prompt, as well as the newsletter subscriber's information. The `configurationId` matches the `configurationId provided to the publisher from Google, in response to the per-newsletter configuration authored by the publisher in the initial step.
+
+```javascript
+{
+  'configurationId': '123-456-789',
+  'data': {
```
Comment (mhkawano) [inline app/content/newsletter-manual.md:170]: We want to split the name into "displayName", "givenName", and "familyName"

File: app/content/newsletter-automatic.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  eventManager.registerEventListener(console.log);
+
+  // Get all available interventions
+  const availableInterventions =
+    await subscriptions.getAvailableInterventions();
+
+  console.log({availableInterventions});
+
+  // Return the prompt, if available, for the current configurationId
+  const prompt = availableInterventions.find(({intervention}) => {
```
Comment (mhkawano) [inline app/content/newsletter-automatic.md:27]: you don't want to destructure intervention

Comment (mhkawano) [reply app/content/newsletter-automatic.md:27]: Same with `{availableInterventions}` and other places

File: app/views/layouts/demo-layout.html [hunk 5]
```diff
@@ -15,7 +15,7 @@
  limitations under the License.
 -->
 
-<html lang="en">
+<html lang="{{#unless data.options.lang}}en{{else}}{{data.options.lang}}{{/unless}}">
```
Comment (mhkawano) [inline app/views/layouts/demo-layout.html:18]: I don't think this should work right now, should it? We have this pr in progress that will be able to make use of it https://github.com/subscriptions-project/swg-js/pull/3527

**Review by mhkawano — APPROVED** [review]
LGTM

File: app/views/layouts/demo-layout.html (hunk 5 above)
Comment (everyplace) [reply app/views/layouts/demo-layout.html:18]: I was able to get the prompts to render in a specified language only if I changed my browser setting. You are right though, this should work eventually, hopefully after your referenced PR is merged.

File: app/content/newsletter-manual.md [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+To invoke a newsletter prompt, a publisher must use the `configurationId` provided by
+Google in response to submitting a prompt configuration. Publishers use the
+`configurationId` to fetch a valid prompt instance using the `subscriptions.getAvailableInterventions()` method from the initialized `swg.js` library.
+
+```javascript
+const configurationId = '<id returend after submitting a newsletter config>';
+
+const availableInterventions = await subscriptions.getAvailableInterventions();
+
+const prompt = availableInterventions.find(({intervention}) => {
```
Comment (mhkawano) [inline app/content/newsletter-manual.md:177]: nit: same issue

**Review by mhkawano — DISMISSED** [review]

**Review by mhkawano — APPROVED** [review]

**Pull request merged**